climt/_components/grid_scale_condensation.py

```diff
@@ -24,55 +24,23 @@ class GridScaleCondensation(Implicit):
         'air_temperature', 'specific_humidity',
     )
 
-    def __init__(self,
-                 gas_constant_of_dry_air=None,
-                 gas_constant_of_water_vapor=None,
-                 heat_capacity_of_dry_air_at_constant_pressure=None,
-                 latent_heat_of_vaporization_of_water=None,
-                 gravitational_acceleration=None,
-                 density_of_liquid_water=None):
+    def __init__(self):
         """
-
-        Args:
-            gas_constant_of_dry_air (float, optional):
-                Value in  $J kg^{-1} K^{-1}$.
-                Default taken from :code:`sympl.default_constants`.
-
-            gas_constant_of_water_vapor (float, optional):
-                Value in $J kg^{-1} K^{-1}$.
-                Default taken from :code:`sympl.default_constants`.
-
-            heat_capacity_of_dry_air_at_constant_pressure (float, optional):
-                Value in $J kg^{-1} K^{-1}$.
-                Default taken from :code:`sympl.default_constants`.
-
-            latent_heat_of_vaporization_of_water (float, optional):
-                Value in $J kg^{-1}$.
-                Default taken from :code:`sympl.default_constants`.
-
-            gravitational_acceleration (float, optional):
-                Value in $m s^{-2}$. Default taken from :code:`sympl.default_constants`.
-
-            density_of_liquid_water (float, optional):
-                Value in $kg m^{-3}$. Default taken from :code:`sympl.default_constants`.
+        Initialise component.
```

Comment:
No changes necessary here, but something to think about: Should we be saying what constants the object uses in the docstrings? Maybe not.

Reply:
Yes, I don't see any obvious need to document constants used. What worries me, however, is that if someone modifies the constants dict _after_ creating the objects, and then goes on the create additional objects (in an untidy jupyter notebook)...

Reply:
You can avoid that by retrieving the constants in the __call__ method instead of in __init__. I think that would be best practice.

Reply:
hmm yes. I will put this on the to-do list for the next release.